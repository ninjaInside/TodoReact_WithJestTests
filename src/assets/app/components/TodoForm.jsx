import React, { useRef } from 'react'

function TodoForm(props) {
	const inputTodo = useRef(null)

	return (
		<>
			<input ref={inputTodo} />
			<button onClick={() => {
				props.addTodo(inputTodo.current.value)
				inputTodo.current.value = ''
			}}>Set todo</button>
		</>
	)
}

export default TodoForm