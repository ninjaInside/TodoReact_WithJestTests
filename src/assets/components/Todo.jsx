import React, { useRef } from 'react'

import styles from '../styles/main.sass'

function Todo(props) {
	let todo = useRef(null)

	return (
		<li className={styles.todoItem} ref={todo}>
			<span className={styles.todoItem__text}>{props.value}</span>
			<button 
				className={styles.todoItem__btn}
				onClick={() => {
					todo.current.remove()
				}}>C</button>
		</li>
	)
}

export default Todo