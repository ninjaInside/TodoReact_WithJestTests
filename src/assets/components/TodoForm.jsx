import React, { useRef } from 'react'

import styles from '../styles/main.sass'

function TodoForm(props) {
	const inputTodo = useRef(null)

	return (
		<>
			<input 
				ref={inputTodo}
				className={styles.todoField__inputAdd}
				placeholder='Add Todo'/>
			<button 
				className={styles.todoField__btn}
				onClick={() => {
					props.addTodo(inputTodo.current.value)
					inputTodo.current.value = ''}}>Set todo</button>
		</>
	)
}

export default TodoForm