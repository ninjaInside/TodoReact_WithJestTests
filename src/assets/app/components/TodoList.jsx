import React from 'react'

import Todo from 'Components/Todo.jsx'

import styles from 'Styles/main.sass'

function TodoList(props) {
	return (
		<ul className={styles.todoField__todosList}>
			{props.list.map((todo, index) => <Todo value={todo} key={index} />)}
		</ul>
	)
}

export default TodoList