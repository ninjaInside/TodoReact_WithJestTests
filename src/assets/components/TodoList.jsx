import React from 'react'

import Todo from 'Components/Todo.jsx'

import styles from '../styles/main.sass'

function TodoList(props) {
	return (
		<ul className={styles.todoField__todosList}>
			{props.list.map((todo, index) => <Todo value={todo} key={index} />)}
		</ul>
	)
}

export default TodoList