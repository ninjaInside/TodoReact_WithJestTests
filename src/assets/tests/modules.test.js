import React from 'react'
import renderer from 'react-test-renderer'

import Todo from '../components/Todo.jsx'
import TodoList from '../components/TodoList.jsx'
import TodoForm from '../components/TodoForm.jsx'

test(`Test "Todo.jsx" component`, () => {
	let component = renderer.create(<Todo value='todos' />)

	let tree = component.toJSON()

	expect(tree).toMatchSnapshot()
})

test(`Test "TodoList.jsx" component`, () => {
	let component = renderer.create(<Todo value='todos' />)

	let tree = component.toJSON()
	
	expect(tree).toMatchSnapshot()
})

test(`Test "TodoForm.jsx" component`, () => {
	let component = renderer.create(<Todo value='todos' />)

	let tree = component.toJSON()
	
	expect(tree).toMatchSnapshot()
})