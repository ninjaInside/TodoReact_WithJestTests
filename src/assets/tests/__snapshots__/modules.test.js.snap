// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test "Todo.jsx" component 1`] = `
<li>
  todos
</li>
`;

exports[`Test "Todo.jsx" components 1`] = `
<li>
  todos
</li>
`;

exports[`Test "TodoForm.jsx" component 1`] = `
<li>
  todos
</li>
`;

exports[`Test "TodoForm.jsx" components 1`] = `
<li>
  todos
</li>
`;

exports[`Test "TodoList.jsx" component 1`] = `
<li>
  todos
</li>
`;

exports[`Test "TodoList.jsx" components 1`] = `
<li>
  todos
</li>
`;
